sdl2promela/stop_condition/translator.py

```diff
@@ -708,13 +762,122 @@ def _generate_queue_length_call(context: GenerateContext, expr: model.CallExpres
     )
 
 
+def _generate_get_state_call(context: GenerateContext, expr: model.CallExpression):
+    if len(expr.parameters) != 1:
```

Comment:
Is there any chance to refactor this method (e.g., split into smaller ones?)

Reply:
refactored